src/pages/iou/MoneyRequestModal.js

```diff
@@ -0,0 +1,499 @@
+    iou: {
+        creatingIOUTransaction: false,
+        error: false,
+        selectedCurrencyCode: null,
+    },
+};
+
+// Determines type of step to display within Modal, value provides the title for that page.
+const Steps = {
+    IOUAmount: 'iou.amount',
+    IOUParticipants: 'iou.participants',
+    IOUConfirm: 'iou.confirm',
+};
+
+const MoneyRequestModal = (props) => {
+    const reportParticipants = lodashGet(props, 'report.participants', []);
+    const participantsWithDetails = _.map(OptionsListUtils.getPersonalDetailsForLogins(reportParticipants, props.personalDetails), personalDetails => ({
+        login: personalDetails.login,
+        text: personalDetails.displayName,
+        firstName: lodashGet(personalDetails, 'firstName', ''),
+        lastName: lodashGet(personalDetails, 'lastName', ''),
+        alternateText: Str.isSMSLogin(personalDetails.login) ? Str.removeSMSDomain(personalDetails.login) : personalDetails.login,
+        icons: [{
+            source: ReportUtils.getAvatar(personalDetails.avatar, personalDetails.login),
+            name: personalDetails.login,
+            type: CONST.ICON_TYPE_AVATAR,
+        }],
+        keyForList: personalDetails.login,
+        payPalMeAddress: lodashGet(personalDetails, 'payPalMeAddress', ''),
+        phoneNumber: lodashGet(personalDetails, 'phoneNumber', ''),
+    }));
+
+    // Skip IOUParticipants step if participants are passed in
+    const steps = reportParticipants.length ? [Steps.IOUAmount, Steps.IOUConfirm] : [Steps.IOUAmount, Steps.IOUParticipants, Steps.IOUConfirm];
+
+    const prevCreatingIOUTransactionStatusRef = useRef();
+    const prevNetworkStatusRef = useRef();
+    const prevSelectedCurrencyCodeRef = useRef();
+
+    const [previousStepIndex, setPreviousStepIndex] = useState(0);
+    const [currentStepIndex, setCurrentStepIndex] = useState(0);
+    const [participants, setParticipants] = useState(participantsWithDetails);
+    const [amount, setAmount] = useState('');
+    const [comment, setComment] = useState('');
+
+    useEffect(() => {
+        PersonalDetails.openMoneyRequestModalPage();
+        IOU.setIOUSelectedCurrency(props.currentUserPersonalDetails.localCurrencyCode);
+    }, []);
+
+    useEffect(() => {
+        prevCreatingIOUTransactionStatusRef.current = lodashGet(props, 'iou.creatingIOUTransaction');
+    }, [props.iou]);
+
+    useEffect(() => {
+        prevNetworkStatusRef.current = props.network.isOffline;
+    }, [props.network.isOffline]);
+
+    useEffect(() => {
+        prevSelectedCurrencyCodeRef.current = lodashGet(props, 'iou.selectedCurrencyCode');
+    }, [props.iou.selectedCurrencyCode]);
+
+    useEffect(() => {
+        const isDoneCreatingIOUTransaction = prevCreatingIOUTransactionStatusRef.current && !lodashGet(props, 'iou.creatingIOUTransaction');
+
+        // User came back online, so we can try to create the IOU transaction again
+        if (prevNetworkStatusRef.current && !props.network.isOffline) {
```

Comment:
NAB / maybe for a follow up. But I would consider refactoring this logic into a custom hook called `useNetworkReconnect()` that does something like this:

exports the context object from here: https://github.com/Expensify/App/blob/4efde840cc7456d14a2f79185e95cd9d95f79b90/src/components/OnyxProvider.js#L9-L10

```js
function useNetworkReconnect(callback) {
    const {isOffline} = useContext(NetworkContext);
    const prevIsOfflineRef = useRef(isOffline);
    useEffect(() => {
        if (prevIsOfflineRef.current && !isOffline) {
            callback();
        }

        prevIsOfflineRef.current = isOffline;
    }, [isOffline]);
}
```

and would be used like this

```js
useNetworkReconnect(PersonalDetails.openMoneyRequestModalPage);
```

Reply:
I agree it would be nice to have a seperate hook for this, I'll create another issue that will go in and just refactor all instances of it

Reply:
https://github.com/Expensify/App/issues/16889